testproject/Assets/Tests/Manual/Scripts/RandomMovement.cs

```diff
@@ -1,19 +1,34 @@
 using UnityEngine;
+using MLAPI;
 
 namespace TestProject.ManualTests
 {
     /// <summary>
     /// Used with GenericObjects to randomly move them around
     /// </summary>
-    public class RandomMovement : MonoBehaviour, IPlayerMovement
+    public class RandomMovement : NetworkBehaviour, IPlayerMovement
     {
         private Vector3 m_Direction;
         private Rigidbody m_Rigidbody;
 
-        public void Start()
+
+        public override void NetworkStart()
         {
             m_Rigidbody = GetComponent<Rigidbody>();
-            ChangeDirection(true, true);
+            if (NetworkObject != null)
+            {
+                if (!NetworkObject.IsOwner)
```

Comment:
`NetworkBehaviour.IsOwner` property does exist :)

Reply:
so `m_Rigidbody.isKinematic = !IsOwner;` would be nice!

Reply:
You are right... that is cleaner for sure.